cmd/xl-v1-metadata.go

```diff
@@ -47,91 +124,148 @@ func (t byObjectPartNumber) Len() int           { return len(t) }
 func (t byObjectPartNumber) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
 func (t byObjectPartNumber) Less(i, j int) bool { return t[i].Number < t[j].Number }
 
-// checkSumInfo - carries checksums of individual scattered parts per disk.
-type checkSumInfo struct {
-	Name      string   `json:"name"`
-	Algorithm HashAlgo `json:"algorithm"`
-	Hash      string   `json:"hash"`
+// ChecksumInfo - carries checksums of individual scattered parts per disk.
+type ChecksumInfo struct {
+	Name string
+	Hash []byte
 }
 
-// HashAlgo - represents a supported hashing algorithm for bitrot
-// verification.
-type HashAlgo string
+// MarshalJSON marshals the ChecksumInfo struct
+func (c ChecksumInfo) MarshalJSON() ([]byte, error) {
+	type checksuminfo struct {
+		Name string `json:"name"`
+		Hash string `json:"hash"`
+	}
 
-const (
-	// HashBlake2b represents the Blake 2b hashing algorithm
-	HashBlake2b HashAlgo = "blake2b"
-	// HashSha256 represents the SHA256 hashing algorithm
-	HashSha256 HashAlgo = "sha256"
-)
+	info := checksuminfo{
+		Name: c.Name,
+		Hash: hex.EncodeToString(c.Hash),
+	}
+	return json.Marshal(info)
+}
+
+// UnmarshalJSON unmarshals the the given data into the ChecksumInfo struct
+func (c *ChecksumInfo) UnmarshalJSON(data []byte) error {
+	type checksuminfo struct {
```

Comment:
redeclaration of struct (from line 135-138) -- can we refactor it to a single struct?